kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemMetadataProvider.java

```diff
@@ -534,6 +535,16 @@ static void writeDescriptor(
         throw new DatasetIOException("Cannot close", e);
       }
     }
+
+    //rename the file into its final location so that changes appear atomic
+      try {
+          fs.rename(tempDescriptorPath, descriptorPath);
```

Comment:
@ben-roling and I were chatting about this this morning. FileContext would do the rename you're expecting but wouldn't pass on the hadoop-1 profile. We're seeing that issue over on another PR now https://github.com/kite-sdk/kite/pull/348#discussion_r28332266